src/tiles/image-block.ts

```diff
@@ -51,35 +52,42 @@ export class ImageBlock extends LitElement {
   }
 
   private get iconOverlayTemplate() {
+    // Only list tiles use the icon overlay
     if (!this.isListTile) return nothing;
 
-    if (!this.model?.loginRequired && !this.model?.contentWarning)
-      return nothing;
+    const { overlayType } = this;
+    if (!overlayType) return nothing;
 
     return html`
       <icon-overlay
-        .loggedIn=${this.loggedIn}
-        .loginRequired=${this.model?.loginRequired}
-        .isCompactTile=${this.isCompactTile}
+        class=${this.isCompactTile ? 'list-compact' : 'list-detail'}
+        .type=${this.overlayType}
       >
       </icon-overlay>
     `;
   }
 
   private get textOverlayTemplate() {
+    // List tiles do not require the text overlay
     if (this.isListTile) return nothing;
 
-    if (!this.model?.loginRequired && !this.model?.contentWarning)
-      return nothing;
+    const { overlayType } = this;
+    if (!overlayType) return nothing;
 
-    return html`
-      <text-overlay
-        .loggedIn=${this.loggedIn}
-        .loginRequired=${this.model?.loginRequired}
-        ?iconRequired=${true}
-      >
-      </text-overlay>
-    `;
+    return html` <text-overlay .type=${this.overlayType}></text-overlay> `;
```

Comment:
QUESTION: The `?iconRequired` attribute is no longer needed?

Reply:
Yes, this property was seemingly unused and perhaps a remnant of some earlier version of the component. Removed it entirely.